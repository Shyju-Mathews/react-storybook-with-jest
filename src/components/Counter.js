import React, { useState } from "react";

const Counter = () => {
  const [count, setCount] = useState(0);

  const handleClick = (cmd) => {
    if (cmd === "dec") {
      // if (count === 0) return;
      setCount((prev) => prev - 1);
    } else if (cmd === "inc") {
      setCount((prev) => prev + 1);
    }
  };
  return (
    <div>
      <h1 data-testid="counter-name" className="text-center mb-3">
        Counter App
      </h1>
      <div className="inline-flex items-center w-full justify-center gap-3">
        <button
          data-testid="decCount"
          className="bg-blue-500 text-white px-4 py-2 rounded"
          onClick={() => handleClick("dec")}
        >
          -
        </button>
        <div
          data-testid="count"
          className="bg-gray-200 px-2 py-1 rounded text-lg"
        >
          {count}
        </div>
        <button
          data-testid="incCount"
          className="bg-blue-500 text-white px-4 py-2 rounded"
          onClick={() => handleClick("inc")}
        >
          +
        </button>
      </div>
    </div>
  );
};

export default Counter;
