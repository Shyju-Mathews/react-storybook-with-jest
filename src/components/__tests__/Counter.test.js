import { render, fireEvent, screen } from "@testing-library/react"
import Counter from "../Counter"
import renderer from "react-test-renderer"

// Method 1

// test('Counter', () => {
//     render(<Counter />);
//     const counterName = screen.getByTestId("counter-name")
//     expect(counterName).toBeInTheDocument();

// const count = screen.getByTestId("count")
// const increment = screen.getByTestId("incCount")

// // interact with elements

// fireEvent.click(increment)
// expect(count).toHaveTextContent("1")

//   });


// Method 2

  describe('Counter', () => {

    // function 1

    it("should inc counter", () => {
        render(<Counter />);
    
    const count = screen.getByTestId("count")
    const increment = screen.getByTestId("incCount")
    
    
    fireEvent.click(increment)
    
    expect(count).toHaveTextContent("1")
      })
    
    //   function 2

    it("should dec counter", () => {
        render(<Counter />);
    
    const count = screen.getByTestId("count")
    const decrement = screen.getByTestId("decCount")
    
    
    fireEvent.click(decrement)
    
    expect(count).toHaveTextContent("-1")
      })
  })

//   snapshot

describe('Counter Snapshot', () => {
  it("should matches DOM Snapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

  



