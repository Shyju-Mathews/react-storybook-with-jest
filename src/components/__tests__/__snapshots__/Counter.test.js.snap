// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot should matches DOM Snapshot 1`] = `
<div>
  <h1
    className="text-center mb-3"
    data-testid="counter-name"
  >
    Counter App
  </h1>
  <div
    className="inline-flex items-center w-full justify-center gap-3"
  >
    <button
      className="bg-blue-500 text-white px-4 py-2 rounded"
      data-testid="decCount"
      onClick={[Function]}
    >
      -
    </button>
    <div
      className="bg-gray-200 px-2 py-1 rounded text-lg"
      data-testid="count"
    >
      0
    </div>
    <button
      className="bg-blue-500 text-white px-4 py-2 rounded"
      data-testid="incCount"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
